tests/core/test_external_search.py

```diff
@@ -1306,8 +1306,10 @@ def test_remove_work(self, end_to_end_search_fixture: EndToEndSearchFixture):
         end_to_end_search_fixture.populate_search_index()
         search.remove_work(data.moby_dick)
         search.remove_work(data.moby_duck)
-        # Immediately querying never works, the search index needs a second to refresh its cache/index/data
-        time.sleep(1)
+
+        # Immediately querying never works, the search index needs to refresh its cache/index/data
+        search.indices.refresh()
```

Comment:
This seems to solve the problem. You could also get similar behavior from the fixture with:
```
end_to_end_search_fixture.external_search.refresh_and_wait()
```

Not sure which would be preferable.

Reply:
They're both identical in the current us case since `refresh_and_wait` uses the `/_refresh` API, which is the same as the `refresh()` API method.